src/Internal/Test/E2E/Runner.purs

```diff
@@ -303,26 +296,73 @@ testPlan opts@{ tests } rt@{ wallets } =
                   , confirmAccess: confirmAccess extensionId re
                   , sign: sign extensionId password re
                   }
-              makeAff $ \k -> do
+
+              subscribeToBrowserEvents page \waitNextEvent -> do
                 let
-                  rethrow aff = launchAff_ do
-                    res <- try aff
-                    when (isLeft res) $ liftEffect $ k res
-                map fiberCanceler $ launchAff $ (try >=> k >>> liftEffect) $
-                  subscribeToBrowserEvents page
-                    case _ of
-                      ConfirmAccess -> rethrow someWallet.confirmAccess
-                      Sign -> rethrow someWallet.sign
+                  handler event = do
+                    case event of
+                      ConfirmAccess -> do
+                        handler =<< raceMany
+                          [ someWallet.confirmAccess *> never
+                          , waitNextEvent
+                          , delaySec (Seconds 100.0) *>
+                              throwError (error nextAfterConfirmAccess)
+                          ]
```

Comment:
This effectively sets 100 seconds timeout for every test. It's not what we want, the test suite should control the timeouts, not the testing engine.